pandas/core/frame.py

```diff
@@ -8441,6 +8443,10 @@ def blk_func(values):
             assert isinstance(res, dict)
             if len(res):
                 assert len(res) == max(list(res.keys())) + 1, res.keys()
+            elif not out_dtype:
+                # The default dtype for empty Series will be 'object' instead of
+                #  'float64' in a future version.
```

Comment:
do we have tests which currently hit this case?

Reply:
this was added because of test failures in tests/frame/test_analytics.py (maybe others)

```
========================================================================= short test summary info ==========================================================================
FAILED pandas/tests/frame/test_analytics.py::TestDataFrameAnalytics::test_stat_op_calc - AssertionError: Caused unexpected warning(s): [('DeprecationWarning', Deprecation...
FAILED pandas/tests/frame/test_analytics.py::TestDataFrameAnalytics::test_median - AssertionError: Caused unexpected warning(s): [('DeprecationWarning', DeprecationWarnin...
FAILED pandas/tests/frame/test_analytics.py::TestDataFrameAnalytics::test_mean_excludes_datetimes[None] - AssertionError: Caused unexpected warning(s): [('DeprecationWarn...
FAILED pandas/tests/frame/test_analytics.py::TestDataFrameAnalytics::test_mean_excludes_datetimes[UTC] - AssertionError: Caused unexpected warning(s): [('DeprecationWarni...
```

Reply:
> do we have tests which currently hit this case?

so I guess on master, no tests hit this case

Reply:
these cases were for empty DataFrames (after removing timestamp columns). using all for is_numeric was returning True for empty iterable (_mgr.blocks). changing is_numeric and this elif not out_dtype code block doesn't need to be added.